File: TestAPI/API_Test_Script.js
const axios = require('axios');

const baseUrl = 'http://localhost:3000/masks';

async function checkApiStatus(url) {
    try {
        const response = await axios.get(url);
        console.log("200 OK.\n");
    } catch (error) {
        console.log(`Problème d'accès à l'API, status ${error.response.status}.`);
    }
}

async function getData(url) {
    try {
        const response = await axios.get(url);
        console.log('GET Response:', response.data);
    } catch (error) {
        console.log('GET Error:', error.message);
    }
}

async function postData(url, data) {
    try {
        const response = await axios.post(url, data);
        console.log('POST Response:', response.data);
        return response.data.id;
    } catch (error) {
        console.log('POST Error:', error.message);
    }
}

async function putData(url, data) {
    try {
        const response = await axios.put(url, data);
        console.log('PUT Response:', response.status);
    } catch (error) {
        console.log('PUT Error:', error.message);
    }
}

async function deleteData(url) {
    try {
        const response = await axios.delete(url);
        console.log('DELETE Response:', response.status);
    } catch (error) {
        console.log('DELETE Error:', error.message);
    }
}

async function main() {
    await checkApiStatus(baseUrl);

    const dataPost = {
        description: "document de test pour le post",
        name: "TEST_POST",
        mask_json: {}
    };

    const dataPut = {
        description: "document de test pour le put",
        name: "TEST_PUT",
        mask_json: {}
    };

    const itemId = await postData(baseUrl, dataPost);
    if (itemId) {
        const itemUrl = `${baseUrl}/${itemId}`;
        await getData(itemUrl);
        await putData(itemUrl, dataPut);
        await deleteData(itemUrl);
        await getData(baseUrl);
    } else {
        console.log('Erreur lors de la création de l\'élément.');
    }
}

main();

File: TestAPI/API_Test_Script.py
import requests

BASE_URL = 'http://localhost:3000/masks'

def check_api_status(url):
    """Check du statut de l'API"""
    response = requests.get(url)
    if response.status_code == 200:
        print('200 OK.')
    else:
        print(f"Problème d'accès à l'API, status {response.status_code}.")

def get_data(url):
    """requête GET"""
    response = requests.get(url)
    print('GET Response:', response.json())

def post_data(url, data):
    """requête POST"""
    response = requests.post(url, json=data)
    print('POST Response:', response.json())
    # Retourner l'ID de l'élément créé
    return response.json().get('id')

def put_data(url, data):
    """requête PUT"""
    response = requests.put(url, json=data)
    print('PUT Response:', response.status_code)

def delete_data(url):
    """requête DELETE"""
    response = requests.delete(url)
    print('DELETE Response:', response.status_code)

check_api_status(BASE_URL)

data_post = {
    "description": "document de test pour le post",
    "name": "TEST_POST",
    "mask_json": {}
}

data_put = {
    "description": "document de test pour le put",
    "name": "TEST_PUT",
    "mask_json": {}
}

item_id = post_data(BASE_URL, data_post)

if item_id:
    url = f"{BASE_URL}/{item_id}"
    get_data(url)
    put_data(url, data_put)
    delete_data(url)
    get_data(BASE_URL)
else:
    print("Erreur lors de la récupération de l'ID après POST.")
